micrometer-core/src/main/java/io/micrometer/core/instrument/config/MeterFilter.java

```diff
@@ -301,7 +300,7 @@ static MeterFilter maxExpected(String prefix, Duration max) {
         return new MeterFilter() {
             @Override
             public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
-                if (Meter.Type.TIMER.equals(id.getType()) && id.getName().startsWith(prefix)) {
+                if (id.getType() == Meter.Type.TIMER && id.getName().startsWith(prefix)) {
```

Comment:
This change removes an inherent null-check, but Id's `type` is not marked as Nullable, so it is probably okay.

Reply:
I'm not sure I understood your comment correctly but it should work as before regardless its nullability. Am I missing something?

Reply:
Sorry, I was wrong. You're right, it will work the same.